src/Core/Entities.Exercises/ExerciseRoutine.cs

```diff
@@ -1,25 +1,50 @@
+using Core.Entities.Base;
+
 namespace Core.Entities.Exercises;
 
-public record ExerciseRoutine
+public class ExerciseRoutine : BaseEntity<Guid>
 {
+    private HashSet<WorkoutPlanStep> _routineSteps = null!;
+
     private ExerciseRoutine()
     {
         
     }
     public ExerciseRoutine(Exercise exercise, int sets, int reps, TimeSpan restTime)
```

Comment:
misschien is het interessant om van sets, reps en resttime eigen value objects te maken, op die manier kunnen ze niet door elkaar gehaald worden en verhuist de validatie naar dat object zelf. Eventueel kan sets en reps in 1 object want lijkt me aan elkaar gerelateerd? Extra voordeel dat de aanroepende code leesbaarder word (maar ook drukker, al kan dat opgelost worden met implicit operators)